app/components/work_package_relations_tab/work_package_relation_form_component.rb

```diff
@@ -48,6 +48,13 @@ def initialize(work_package:, relation:, base_errors: nil)
 
   def related_work_package
     @related_work_package ||= @relation.to
+
+    # We cannot rely on the related WorkPackage being the "to", depending on the relation it can also be "from"
+    if @related_work_package.id == @work_package.id
+      @related_work_package = @relation.from
+    end
+
+    @related_work_package
   end
```

Comment:
This works, but the memozation that was used in line 50 is no longer in-effect here. `@related_work_packages` will be re-assigned for this object whenever the `from` relation is used.

To keep the memoization intact, we could alter the code like this:

```ruby
def related_work_package
  @related_work_package ||= begin
    related = @relation.to
    # We cannot rely on the related WorkPackage being the "to", depending on the relation it can also be "from"
    related.id == @work_package.id ? @relation.from : related
  end
end
```

This will execute the code only on the first call to `related_work_package`. Subsequent method calls will immediately return the memoized value :) What do you think?

Reply:
Awesome, thanks a lot 👍